sdk/servicefabricmanagedclusters/Azure.ResourceManager.ServiceFabricManagedClusters/src/Azure.ResourceManager.ServiceFabricManagedClusters.csproj

```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <Version>1.3.0-beta.1</Version>
     <!--The ApiCompatVersion is managed automatically and should not generally be modified manually.-->
-    <ApiCompatVersion>1.2.0</ApiCompatVersion>
```

Comment:
I see a bunch of these errors when I don't comment it out. How can I resolve these ?


C:\Users\mwesigwaguma\.nuget\packages\microsoft.dotnet.apicompat\5.0.0-beta.20467.1\build\Microsoft.DotNet.ApiCompat.targets(82,5): error : MembersMustExist : Member '
public System.Nullable<System.Boolean> Azure.ResourceManager.ServiceFabricManagedClusters.ServiceFabricManagedClusterData.IsHttpGatewayExclusiveAuthModeEnabled.get()'
does not exist in the implementation but it does exist in the contract. [E:\repos\azure-sdk-for-net\sdk\servicefabricmanagedclusters\Azure.ResourceManager.ServiceFabri
cManagedClusters\src\Azure.ResourceManager.ServiceFabricManagedClusters.csproj::TargetFramework=netstandard2.0]
C:\Users\mwesigwaguma\.nuget\packages\microsoft.dotnet.apicompat\5.0.0-beta.20467.1\build\Microsoft.DotNet.ApiCompat.targets(82,5): error : MembersMustExist : Member '
public void Azure.ResourceManager.ServiceFabricManagedClusters.ServiceFabricManagedClusterData.IsHttpGatewayExclusiveAuthModeEnabled.set(System.Nullable<System.Boolean
>)' does not exist in the implementation but it does exist in the contract. [E:\repos\azure-sdk-for-net\sdk\servicefabricmanagedclusters\Azure.ResourceManager.ServiceF
abricManagedClusters\src\Azure.ResourceManager.ServiceFabricManagedClusters.csproj::TargetFramework=netstandard2.0]
C:\Users\mwesigwaguma\.nuget\packages\microsoft.dotnet.apicompat\5.0.0-beta.20467.1\build\Microsoft.DotNet.ApiCompat.targets(82,5): error : MembersMustExist : Member '
public Azure.ResourceManager.ServiceFabricManagedClusters.Models.ManagedClusterUpgradePolicy Azure.ResourceManager.ServiceFabricManagedClusters.ServiceFabricManagedClu
sterData.UpgradeDescription.get()' does not exist in the implementation but it does exist in the contract. [E:\repos\azure-sdk-for-net\sdk\servicefabricmanagedclusters
\Azure.ResourceManager.ServiceFabricManagedClusters\src\Azure.ResourceManager.ServiceFabricManagedClusters.csproj::TargetFramework=netstandard2.0]
C:\Users\mwesigwaguma\.nuget\packages\microsoft.dotnet.apicompat\5.0.0-beta.20467.1\build\Microsoft.DotNet.ApiCompat.targets(82,5): error : MembersMustExist : Member '
public void Azure.ResourceManager.ServiceFabricManagedClusters.ServiceFabricManagedClusterData.UpgradeDescription.set(Azure.ResourceManager.ServiceFabricManagedCluster
s.Models.ManagedClusterUpgradePolicy)' does not exist in the implementation but it does exist in the contract. [E:\repos\azure-sdk-for-net\sdk\servicefabricmanagedclus
ters\Azure.ResourceManager.ServiceFabricManagedClusters\src\Azure.ResourceManager.ServiceFabricManagedClusters.csproj::TargetFramework=netstandard2.0]
C:\Users\mwesigwaguma\.nuget\packages\microsoft.dotnet.apicompat\5.0.0-beta.20467.1\build\Microsoft.DotNet.ApiCompat.targets(82,5): error : MembersMustExist : Member '
public Azure.Response<Azure.ResourceManager.ServiceFabricManagedClusters.Models.ManagedMaintenanceWindowStatus> Azure.ResourceManager.ServiceFabricManagedClusters.Serv
iceFabricManagedClusterResource.GetManagedMaintenanceWindowStatus(System.Threading.CancellationToken)' does not exist in the implementation but it does exist in the co
ntract. [E:\repos\azure-sdk-for-net\sdk\servicefabricmanagedclusters\Azure.ResourceManager.ServiceFabricManagedClusters\src\Azure.ResourceManager.ServiceFabricManagedC
lusters.csproj::TargetFramework=netstandard2.0]

Reply:
this section says I can remove it and it will be added back : https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#api-compatibility-verification

Reply:
I think the guide is not clear here. For mgmt SDK, we don't allow removal of `<ApiCompat>` unless you have very strong reasons. Those errors mean you have incompatible changes in the interface of the new release. To work around, you need to write customization codes. You can take a look at how `compute` resolves those issues: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/compute/Azure.ResourceManager.Compute/src/Customize

There are also some guides here: https://github.com/Azure/autorest.csharp#customizing-the-generated-code